### PR TITLE
fix(ComboBox): programmatically set selected index

### PR DESCRIPTION
**Pull request by patoi**
The last release (https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.51.2) created a regression with programmatically set selected index, this PR fix it.

### CODE REVIEW

**Comment by patoi** [comment]
https://user-images.githubusercontent.com/1846548/148940527-8e2e2ecc-7b3e-406c-8df3-ee72ab5f87c8.mp4

File: docs/src/pages/components/ComboBox.svx [hunk 1]
```diff
@@ -3,7 +3,7 @@
   import { Button } from "carbon-components-svelte";
   import Preview from "../../components/Preview.svelte";
   let comboboxComponent
-
+  let selectedIndex = 1
```
Comment (metonym) [inline docs/src/pages/components/ComboBox.svx:6]: Is this line needed?

File: docs/src/pages/components/ComboBox.svx [hunk 2]
```diff
@@ -25,7 +25,12 @@ items={[
   {id: "2", text: "Fax"}
   ]}  />
 
+### Programmatically set selected index
```
Comment (metonym) [inline docs/src/pages/components/ComboBox.svx:28]: Let's call this "Reactive example"

**Comment by metonym** [comment]
@patoi Could you describe the regression?

**Comment by patoi** [comment]
Regression:

Programmatically set selected index to -1:
- expected: selectedIndex = -1, and inputValue is empty
- actual:  selectedIndex = -1, and inputValue doesn't change

https://user-images.githubusercontent.com/1846548/149087547-12319393-eee1-4cf0-a12f-eb3d0a290530.mp4

File: docs/src/pages/components/ComboBox.svx (hunk 1 above)
Comment (patoi) [reply docs/src/pages/components/ComboBox.svx:6]: I think it is necessary because of `bind:selecetedIndex`, it's a two-way binded value.

**Review by metonym — APPROVED** [review]

**Comment by metonym** [comment]
Released in [v0.51.3](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.51.3)

**Pull request merged**